ooniprobe/View/RunTestViewController.mm

```diff
@@ -10,23 +10,28 @@ @implementation RunTestViewController
 - (void)viewDidLoad {
     [super viewDidLoad];
     [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(reloadTestUI) name:@"reloadTable" object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(reloadTest:) name:@"reloadTest" object:nil];
+    [self configureView];
+}
 
+-(void)configureView{
     currentTests = [Tests currentTests];
     [self.runButton setTitle:[NSString stringWithFormat:@"   %@   ", NSLocalizedString(@"run", nil)] forState:UIControlStateNormal];
-
+    
     if (testDescription != nil)
         [self.titleLabel setText:[NSString stringWithFormat:@"%@", testDescription]];
     else
         [self.titleLabel setText:NSLocalizedString(@"run_test_message", nil)];
-
+    
     [self.test_detailsLabel setText:NSLocalizedString(@"test_details", nil)];
     NetworkMeasurement *current = [currentTests getTestWithName:testName];
     [self.test_titleLabel setText:NSLocalizedString(current.name, nil)];
     [self.test_iconImage setImage:[UIImage imageNamed:[NSString stringWithFormat:@"%@_big", current.name]]];
     
     self.tableView.estimatedRowHeight = 44.0;
     self.tableView.rowHeight = UITableViewAutomaticDimension;
-
+    
+    rows = [[NSArray alloc] init];
```

Comment:
How come this variable was not defined previously? Is this what the fix is? That basically you were re-using every time the same variable?

Reply:
It's initialized below https://github.com/TheTorProject/ooniprobe-ios/blob/e858d8efb75c55b6854d9860f1bdf810c54e5be2/ooniprobe/View/RunTestViewController.mm#L38

This is needed to "reset" the variable when the screen is reopened without closing it.

Reply:
@lorenzoPrimi maybe it helps a comment explaining why you're re-initializing it.

Reply:
to be sure that the rows var is resetted when I open a new screen over the current one

Reply:
Yes yes, that was clear to me. I was suggesting to put this into a comment.

Reply:
Again, I meant a comment in the code, just above when you reinit the array